src/client/Microsoft.Identity.Client/Platforms/iOS/SystemWebview/ASWebAuthenticationPresentationContextProviderWindow.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AuthenticationServices;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Identity.Client.Platforms.iOS.SystemWebview
+{
+    internal class ASWebAuthenticationPresentationContextProviderWindow : NSObject, IASWebAuthenticationPresentationContextProviding
+    {
+        public UIWindow GetPresentationAnchor(ASWebAuthenticationSession session)
+        {
+            return UIApplication.SharedApplication.KeyWindow;
```

Comment:
>KeyWindow [](start = 51, length = 9)

Is this a value we should let customers choose? Perhaps an additional parameter to the "window" object ?

Reply:
i can look into that.

Reply:
Can be for v2 but something we may want to expose at some point in time. This seems like a super first step. Also what made you chose this value and not other values?

---
In reply to: [331277868](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1414#discussion_r331277868) [](ancestors = 331277868)

Reply:
there weren't a lot of other options and this one worked. iOS team did something sort of similar. but will look into it more